pkg/parse/parse.go

```diff
@@ -2,11 +2,13 @@ package parse
 
```

Comment:
Revert this whole file!

Finding files isnt the responsibility of the parser, it is the responsibility of the `afero.Fs`, I would move all this remote loading implementation details to a new afero.Fs implementation which is backed by some other fs, which tried to download if the file doesnt exist. something like:

```
type FsWithModSupport struct {
    fs afero.Fs // the real filesystem to use for file accesses
    ... // whatever is actually needed to run pkg/mod
}

func (fs *FsWithModSupport) OpenFile(filename string) { // whatever the API actually is
   1) attempt to load the file from fs.fs
   2) if it doesnt exist and it looks like a url do the mod download
   3) shuold now be able to load the file from fs.fs
}
```

An advantage of this is testing the fs can be done much more easily, and it can be swapped out if the user wants to forbid downloads (can be a security vulnerability)